plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSessionProperties.java

```diff
@@ -685,6 +703,13 @@ public static int getParquetBatchSize(ConnectorSession session)
         return session.getProperty(PARQUET_WRITER_BATCH_SIZE, Integer.class);
     }
 
+    public static boolean isParquetOptimizedWriterValidate(ConnectorSession session)
+    {
+        double percentage = session.getProperty(PARQUET_OPTIMIZED_WRITER_VALIDATION_PERCENTAGE, Double.class);
+        checkArgument(percentage >= 0.0 && percentage <= 100.0);
+        return ThreadLocalRandom.current().nextDouble(100) < percentage;
```

Comment:
Extract drawing of random number out of session property getter. As it is now, this method semantics is confusing and it's easy to assume that method result does not change with every call, therefore it can be cached for entire table scan.

Reply:
This is copy-paste of `isOrcOptimizedWriterValidate`.
An explicit getter for `PARQUET_OPTIMIZED_WRITER_VALIDATION_PERCENTAGE` wouldn't make sense as we don't want to use it without applying random sampling.
I don't see cause for confusion as the name of the method and signature `boolean isParquetOptimizedWriterValidate` is different from what a simple getter would have (`double getParquetOptimizedWriterValidationPercentage`).